lib/parser/ruby30.y

```diff
@@ -3071,7 +3071,7 @@ require 'parser'
   end
 
   def endless_method_name(name_t)
-    if name_t[0].end_with?('=')
+    if !%w[== === >= <= !=].include?(name_t[0]) && name_t[0].end_with?('=')
```

Comment:
`[]=` should also be in this list (and who knows what else):
``` ruby
> ObjectSpace.each_object(Class).flat_map(&:instance_methods).uniq.select { |mid| mid.to_s.end_with?('=') && !(mid.to_s[-2] =~ /\w/) }
 => [:===, :==, :!=, :<=, :>=, :[]=]
```

Reply:
wait, or is it a setter? 😄

Reply:
`[]=` will also result in the error on MRI :-)

```ruby
% ruby -ve "def []=(index, value) = 42"
ruby 3.0.1p64 (2021-04-05 revision 0fb782ee38) [x86_64-darwin19]
-e:1: setter method cannot be defined in an endless method definition
def []=(index, value) = 42
```

```ruby
% ruby -ve "def []=(index, value) = 42"
ruby 3.1.0dev (2021-04-06T07:03:20Z master 31ba817887) [x86_64-darwin19]
-e:1: setter method cannot be defined in an endless method definition
def []=(index, value) = 42
```

Reply:
Oh, you're right. It's a setter, so it's an error and it's expected. I updated this PR again.